### PR TITLE
Switch to createDataFrame

### LINKED ISSUE #763: Frontend DS template and tutorial use deprecated `MutableDataFrame`
Warning message: `'MutableDataFrame' is deprecated. use standard DataFrame, or create one with PartialDataFrame`

Example run https://github.com/grafana/plugin-tools/actions/runs/7987513869

And in our docs https://grafana.com/developers/plugin-tools/tutorials/build-a-data-source-plugin#returning-data-frames

Will need to check more docs, saw in Explore queries as well

### PR DESCRIPTION
**Pull request by tolzhabayev**
**What this PR does / why we need it**:
Move from deprecated `MutableDataFrame` to the function the creates `DataFrame`'s

**Which issue(s) this PR fixes**:

Fixes #763

**Special notes for your reviewer**:

### CODE REVIEW

**Review by sunker — APPROVED** [review]
LGTM

File: packages/create-plugin/templates/datasource/src/datasource.ts [hunk 1]
```diff
@@ -77,7 +76,7 @@ export class DataSource extends DataSourceApi<MyQuery, MyDataSourceOptions> {
       }
     } catch (err) {
       let message = '';
-      if (_.isString(err)) {
+      if (typeof err === 'string') {
```
Comment (tolzhabayev) [inline packages/create-plugin/templates/datasource/src/datasource.ts:79]: I couldn't find any other places we are using `lodash` on scaffolding so I removed it - the developers can add it back when needed and it is removed by webpack on bundle anyway.

Comment (tolzhabayev) [reply packages/create-plugin/templates/datasource/src/datasource.ts:79]: @jackw anything against that ☝️

Comment (sunker) [reply packages/create-plugin/templates/datasource/src/datasource.ts:79]: 👍 

This method is a bit of a mess, so would be nice to improve at some point. Maybe grafana/data could expose a function that returns message from an unknown error.

**Pull request merged**